velox/substrait/VeloxToSubstraitExpr.cpp

```diff
@@ -462,6 +495,10 @@ const ::substrait::Expression& VeloxToSubstraitExprConvertor::toSubstraitExpr(
   auto inputs = callTypeExpr->inputs();
   auto& functionName = callTypeExpr->name();
 
+  if (functionName == "in") {
```

Comment:
Hardcoding the name of the function is problematic. This function may be registered under a different name or a different function can be registered under the "in" name.

I wonder if a more robust approach is to introduce Options struct to specify various configurations for the converter and have one of these options be the name of the "IN" function.

Reply:
Understand the issue here. I'm having two questions here:

- How do we know the exact `callTypeExpr->name()` of IN expression in Velox? It could vary according to the prefix.
- Can we record the prefix during registration and append it here?

Reply:
> How do we know the exact callTypeExpr->name() of IN expression in Velox?

We don't. Hence, I suggest we require that the user of the converter specify it.

Reply:
I added prefix in the converter, because this is not just an issue for `IN`. The prefix of other scalar functions also needs to be considered.

Reply:
@rui-mo Rui, I'll take a look, but I don't think it is sufficient to specify he prefix. I imagine we need to specify full function names because it is not guaranteed that "in" name refers to the "right" function.

Reply:
I find presto and spark both register it as `in`, seeing [GeneralFunctionsRegistration.cpp](https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/registration/GeneralFunctionsRegistration.cpp#L24) and [In.cpp](https://github.com/facebookincubator/velox/blob/main/velox/functions/sparksql/In.cpp#L117). How about using certain definition just like aggregate names in [AggregateNames.h](https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/aggregates/AggregateNames.h)?